app/controllers/application_controller/buttons.rb

```diff
@@ -813,38 +812,14 @@ def button_set_resource_action(button)
     end
   end
 
-  def button_build_combo_xml(buttons,b_id)
-    xml = REXML::Document.load("")
-    xml << REXML::XMLDecl.new(1.0, "UTF-8")
-    # Create root element
-    root = xml.add_element("complete")
-    #creating pulldown with 5 buttons
-    opt = root.add_element("option", {"value"=>0})
-    if b_id.blank? || b_id.to_s == "0"
-      opt.add_attribute("selected","true")
-    end
-    #opt.text = "No Button Assigned"
-    opt.text = "<No Image>"
-    temp = Array.new
-    #   buttons.each do |b|
-    #     temp.push(b.button_id) unless temp.include?(b.button_id)
-    #     temp.delete(@edit[:current][:button_id]) if temp.include?(@edit[:current][:button_id]) #deleting the id that's being edited, so it is displayed in combo pull down
-    #   end
-
+  def build_button_image_options
+    options = []
     i = 1
     15.times{
-      if !temp.include?(i)
-        #opt = root.add_element("option", {"value"=>i,"img_src"=>"/images/custom/custom-#{i}.png"})
-        opt = root.add_element("option", {"value"=>i,"img_src"=>"/images/icons/new/custom-#{i}.png"})
-        if i.to_s == b_id.to_s
-          opt.add_attribute("selected","true")
-        end
-        #opt.text = "Button #{i}"
-      end
+      options.push(["Button Image #{i}", i, {"data-icon" => "product product-custom-#{i}"}])
       i+=1
     }
-
-    return xml.to_s
+    options
```

Comment:
This can be simplified to:

```ruby
def build_button_image_options
  15.times.collect { |n| ["Button Image #{i}", i, {"data-icon" => "product product-custom-#{i}"}] }
end
```